tools/tf2openapi/types/tf2openapi_tfmetagraph_test.go

```diff
@@ -0,0 +1,161 @@
+package types
+
+import (
+	"github.com/kubeflow/kfserving/tools/tf2openapi/generated/framework"
+	pb "github.com/kubeflow/kfserving/tools/tf2openapi/generated/protobuf"
+	"github.com/onsi/gomega"
+	"testing"
+)
+
+/* Expected values */
+func expectedTFMetaGraph() TFMetaGraph {
+	return TFMetaGraph{
+		SignatureDefs: []TFSignatureDef{
+			{
+				Key: "sigDefKey",
+				Inputs: []TFTensor{
+					{
+						Name:  "inputTensorName",
+						DType: DtInt8,
+						Shape: TFShape{-1, 3},
+						Rank:  2,
+					},
+				},
+				Outputs: []TFTensor{
+					{
+						Name:  "outputTensorName",
+						DType: DtInt8,
+						Shape: TFShape{-1, 3},
+						Rank:  2,
+					},
+				},
+			},
+		},
+	}
+}
+
+/* Fake protobuf structs to use as test inputs */
+func desiredSigDefPb() *pb.SignatureDef {
+	return &pb.SignatureDef{
+		MethodName: "tensorflow/serving/predict",
+		Inputs: map[string]*pb.TensorInfo{
+			"inputTensorName": {
+				Dtype: framework.DataType_DT_INT8,
+				TensorShape: &framework.TensorShapeProto{
+					Dim: []*framework.TensorShapeProto_Dim{
+						{
+							Size: -1,
+						},
+						{
+							Size: 3,
+						},
+					},
+					UnknownRank: false,
+				},
+			},
+		},
+		Outputs: map[string]*pb.TensorInfo{
+			"outputTensorName": {
+				Dtype: framework.DataType_DT_INT8,
+				TensorShape: &framework.TensorShapeProto{
+					Dim: []*framework.TensorShapeProto_Dim{
+						{
+							Size: -1,
+						},
+						{
+							Size: 3,
+						},
+					},
+					UnknownRank: false,
+				},
+			},
+		},
+	}
+}
+
+func undesiredSigDefPb() *pb.SignatureDef {
+	return &pb.SignatureDef{
+		MethodName: "tensorflow/serving/predict",
+		Inputs: map[string]*pb.TensorInfo{
+			"inputTensorName": {
+				Dtype: framework.DataType_DT_INT8,
+				TensorShape: &framework.TensorShapeProto{
+					Dim: []*framework.TensorShapeProto_Dim{
+						{
+							Size: -1,
+						},
+					},
+					UnknownRank: false,
+				},
+			},
+		},
+		Outputs: map[string]*pb.TensorInfo{
+			"outputTensorName": {
+				Dtype: framework.DataType_DT_INT8,
+				TensorShape: &framework.TensorShapeProto{
+					UnknownRank: true,
+				},
+			},
+		},
+	}
+}
+
+func metaGraphPb() *pb.MetaGraphDef {
+	return &pb.MetaGraphDef{
+		MetaInfoDef: &pb.MetaGraphDef_MetaInfoDef{
+			Tags: []string{
+				"tag",
+			},
+		},
+		SignatureDef: map[string]*pb.SignatureDef{
+			"sigDefKey": desiredSigDefPb(),
+		},
+	}
+}
+
+func metaGraphPbWithMultipleSigDefs() *pb.MetaGraphDef {
+	return &pb.MetaGraphDef{
+		MetaInfoDef: &pb.MetaGraphDef_MetaInfoDef{
+			Tags: []string{
+				"tag",
+			},
+		},
+		SignatureDef: map[string]*pb.SignatureDef{
+			"sigDefKey":          desiredSigDefPb(),
+			"undesiredSigDefKey": undesiredSigDefPb(),
```

Comment:
Does any of the test uses the undesired-ness of this pb? If not I think you do not need to create the `undesiredSigDefPb` method.

As an alternative, in `TestNewTFMetaGraphWithErrSignatureDef`, you could call `NewMetaGraph(metaGraphPbWithMultipleSigDefs, "undesiredSigDefKey")`, so you could avoid the artificial `DType` change.

Reply:
Changing `undesiredSigDefPb` to `badSigDefPb`, which returns a SigDef that will cause an error for use in `TestNewTFMetaGraphWithErrSignatureDef`.